packages/shopify-api/lib/base-types.ts

```diff
@@ -5,32 +5,105 @@ import {AuthScopes} from './auth/scopes';
 import {BillingConfig} from './billing/types';
 import {ApiVersion, LogSeverity} from './types';
 
+/**
+ * A function used by the library to log events related to Shopify.
+ */
 export type LogFunction = (severity: LogSeverity, msg: string) => void;
 
 export interface ConfigParams<
   Resources extends ShopifyRestResources = ShopifyRestResources,
   Future extends FutureFlagOptions = FutureFlagOptions,
 > {
+  /**
+   * The API key for your app.
+   *
+   * You can find this in the Partner Dashboard.
+   */
   apiKey?: string;
+  /**
+   * The API secret key for your app.
+   *
+   * You can find this in the Partner Dashboard.
+   */
   apiSecretKey: string;
+  /**
+   * The scopes your app needs to access the API.
+   */
   scopes?: string[] | AuthScopes;
+  /**
+   * THe host name of your app.
+   */
   hostName: string;
+  /**
+   * The scheme to use for the app host.
+   */
   hostScheme?: 'http' | 'https';
+  /**
+   * The API version to use.
+   */
   apiVersion: ApiVersion;
+  /**
+   * Whether the app is embedded in the Shopify admin.
+   */
   isEmbeddedApp: boolean;
+  /**
+   * Whether the app is a custom store app.
```

Comment:
Could you add a bit more detail here. Is this partner custom apps? or merchant custom apps or both?

These are the documented [app distribution types](https://shopify.dev/docs/apps/distribution). We should reference those.